app/controllers/authentication_controller.rb

```diff
@@ -0,0 +1,12 @@
+class AuthenticationController < ApplicationController
+  def token_auth
+    token = params[:auth_token]
+    payload = GoogleService.decode(token)
+    if User.where('email = ?', payload[:email])
+      render json: {is_new: false, token: token}
+    else
+      User.create!(name: payload[:name], email: payload[:email], google_token: token)
+      render json: {is_new: true, token: token}
+    end
+  end
```

Comment:
Ok, so I think it'll be in here that in addition to rendering the json here, we can set the current_user with info from the user's payload, right? Either way, as long as this authentication has a happy path, we should set the current user. What does the sad path look like?

Reply:
(By making a session)